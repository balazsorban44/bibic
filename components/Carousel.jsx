import React, {useEffect} from 'react'
import Slide from "react-reveal/Slide"
import makeCarousel from "react-reveal/makeCarousel"

const Carousel = ({title, subtitle, data: gallery}) => {

  useEffect(() =>Â {
    window.scrollTo(0, 0)
  }, [])

  const children = gallery
    .map((itemProps, key) =>
      <Slide key={key} right>
        <CarouselItem
          className="carousel-item"
          {...itemProps}
        />
      </Slide>
    )

  return(
    <section
      className="carousel-wrapper"
      id={title}
    >
      <div className="title">
        <h2>{title}</h2>
      </div>
      <CarouselWrapper defaultWait={10000}>
        {children}
      </CarouselWrapper>
      {subtitle ? <h3>{subtitle}</h3> : null}
    </section>
  )
}

export default Carousel

const CarouselWrapper = makeCarousel(({
  position, handleClick, children
}) =>
  <div className="carousel">{children}
    <span
      className="carousel-arrow carousel-arrow-prev"
      data-position={position - 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
    <span
      className="carousel-arrow carousel-arrow-next"
      data-position={position + 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
  </div>)


export const CarouselItem = ({
  title, desc, SIZE_640, SIZE_1024, SIZE_1440, SIZE_ORIGINAL, className
}) =>
  <div className={className}>
    <span>
      <img
        alt="blurred background"
        className="carousel-item-bg"
        src={SIZE_640}
      />
    </span>
    <div className="carousel-item-picture">
      <picture
        className="carousel-item-img"
      >
        <source
          media="(max-width: 640px)"
          srcSet={SIZE_640}
        />
        <source
          media="(max-width: 1024px)"
          srcSet={SIZE_1024}
        />
        <source
          media="(max-width: 1440px)"
          srcSet={SIZE_1440}
        />
        <img
          alt={title}
          src={SIZE_ORIGINAL}
        />
      </picture>
    </div>
    {desc ?
      <div className="carousel-item-desc">
        <h4>{title}</h4>
        <p>{desc}</p>
      </div> : null
    }
  </div>
