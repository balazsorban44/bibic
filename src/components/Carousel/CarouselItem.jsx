import React from "react"


const CarouselItem = ({title, desc, SIZE_640, SIZE_1024, SIZE_1440, SIZE_ORIGINAL, className}) =>
  <div className={className}>
    <span>
      <img
        alt="blurred background"
        className="carousel-item-bg"
        src={SIZE_640}
      />
    </span>
    <div className="carousel-item-picture">
      <picture
        className="carousel-item-img"
      >
        <source
          media="(max-width: 640px)"
          srcSet={SIZE_640}
        />
        <source
          media="(max-width: 1024px)"
          srcSet={SIZE_1024}
        />
        <source
          media="(max-width: 1440px)"
          srcSet={SIZE_1440}
        />
        <img
          alt={title}
          src={SIZE_ORIGINAL}
        />
      </picture>
    </div>
    {desc ?
      <div className="carousel-item-desc">
        <h4>{title}</h4>
        <p>{desc}</p>
      </div> : null
    }
  </div>


export default CarouselItem