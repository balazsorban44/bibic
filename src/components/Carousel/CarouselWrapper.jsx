
import React from "react"
import makeCarousel from "react-reveal/makeCarousel"

const CarouselWrapper = makeCarousel(({position, handleClick, children}) =>
  <div className="carousel">{children}
    <span
      className="carousel-arrow carousel-arrow-prev"
      data-position={position - 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
    <span
      className="carousel-arrow carousel-arrow-next"
      data-position={position + 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
  </div>)

export default CarouselWrapper