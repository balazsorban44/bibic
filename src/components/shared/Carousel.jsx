import React, {Component} from 'react'
import withRouter from 'react-router/withRouter'
import Slide from "react-reveal/Slide"
import makeCarousel from "react-reveal/makeCarousel"
import {withStore} from '../db'
import {Loading} from './Elements'

export class Carousel extends Component {

  componentDidMount() {
    window.scrollTo(0, 0)
  }

  render() {
    let {match: {path}} = this.props
    const {
      title, subtitle, galleries
    } = this.props
    path = path.replace("/", "")
    const children = galleries[path] ?
      Object
        .values(galleries[path])
        .map((itemProps, key) =>
          <Slide
            key={key}
            right
          >
            <CarouselItem
              className="carousel-item"
              {...itemProps}
            />
          </Slide>
        ) :
      <Loading/>
    return(
      <section
        className="carousel-wrapper"
        id={path}
      >
        <div className="title">
          <h2>{title}</h2>
          {subtitle && <h3>{subtitle}</h3> }
        </div>
        <CarouselWrapper defaultWait={10000}>
          {children}
        </CarouselWrapper>
      </section>
    )}
}

const CarouselWrapper = makeCarousel(({
  position, handleClick, children
}) =>
  <div className="carousel">{children}
    <span
      className="carousel-arrow carousel-arrow-prev"
      data-position={position - 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
    <span
      className="carousel-arrow carousel-arrow-next"
      data-position={position + 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
  </div>)

export default withRouter(withStore(Carousel))


export const CarouselItem = ({
  title, desc, SIZE_640, SIZE_1024, SIZE_1440, SIZE_ORIGINAL, className
}) =>
  <div className={className}>
    <span>
      <img
        alt="blurred background"
        className="carousel-item-bg"
        src={SIZE_640}
      />
    </span>
    <div className="carousel-item-picture">
      <picture
        className="carousel-item-img"
      >
        <source
          media="(max-width: 640px)"
          srcSet={SIZE_640}
        />
        <source
          media="(max-width: 1024px)"
          srcSet={SIZE_1024}
        />
        <source
          media="(max-width: 1440px)"
          srcSet={SIZE_1440}
        />
        <img
          alt={title}
          src={SIZE_ORIGINAL}
        />
      </picture>
    </div>
    {desc ?
      <div className="carousel-item-desc">
        <h4>{title}</h4>
        <p>{desc}</p>
      </div> : null
    }
  </div>
