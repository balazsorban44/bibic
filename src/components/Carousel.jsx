import React, {Component, Fragment} from 'react'
import Gallery from './Gallery'
import withRouter from 'react-router/withRouter'
import {BackMenu} from '../Menu'
import {Next, Prev} from './Elements'
import Carousel from 'nuka-carousel'


class Carousel extends Component {

  componentDidMount() {
    window.scrollTo(0, 0)
  }

  render() {
    let {path} = this.props.match
    path = path.replace("/", "")
    return(
      <Fragment>
        <BackMenu/>
        <section id={path}>
          <h2>Ã‰teleink</h2>
          <Gallery
            component={Carousel}
            componentProps={{
              // autoplay: true,
              wrapAround: true,
              className: "foods",
              heightMode: "max",
              cellSpacing: 40,
              renderCenterRightControls: Next,
              renderCenterLeftControls: Prev,
              renderTopRightControls: Next,
              renderTopLeftControls: Prev
            }}
            item={FoodItem}
            path={path}
          />
        </section>
      </Fragment>
    )}
}

export default withRouter(Carousel)


const FoodItem = ({
  title, desc, SIZE_640, SIZE_1280, SIZE_1440, SIZE_ORIGINAL
}) =>
  <div className="food">
    <picture>
      <source
        media="(max-width: 640px)"
        srcSet={SIZE_640}
      />
      <source
        media="(max-width: 1280px)"
        srcSet={SIZE_1280}
      />
      <source
        media="(max-width: 1920px)"
        srcSet={SIZE_1440}
      />
      <img
        alt={title}
        src={SIZE_ORIGINAL}
      />
    </picture>
    <div className="food-desc">
      <h4>{title}</h4>
      <p>{desc}</p>
    </div>
  </div>