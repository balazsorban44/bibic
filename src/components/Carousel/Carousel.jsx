import React from "react"

import Slide from "react-reveal/Slide"

import Loading from "ui/Loading"
import useScrollTo from "hooks/useScrollTo"

import CarouselWrapper from "./CarouselWrapper"
import DefaultCarouselItem from "./CarouselItem"
import useGallery from "hooks/data/useGallery"

export const Carousel = ({type, title, subtitle, className, item, slider=true, count}) => {
  const CarouselItem = item || DefaultCarouselItem

  useScrollTo(0,0)

  const [gallery] = useGallery(type)


  const children =
    Object.values(gallery)
      .slice(0, count)
      .map((item, key) =>
        slider ?
          <Slide key={key} right>
            <CarouselItem className="carousel-item" {...item} />
          </Slide> :
          <CarouselItem className="carousel-item" key={key} {...item} />
      )

  return (
    <div className={className || "carousel-wrapper"} id={type}>
      {title ?
        <div className="title">
          <h2>{title}</h2>
        </div> : null
      }
      {children ?
        slider ?
          <CarouselWrapper defaultWait={10000}>
            {children}
          </CarouselWrapper> :
          children :
        // REVIEW:  turn to component
        <div style={{display: "flex", alignItems: "center", justifyContent: "center"}}>
          <Loading/>
        </div>
      }
      {subtitle ? <h3>{subtitle}</h3> : null }
    </div>
  )
}

export default Carousel

