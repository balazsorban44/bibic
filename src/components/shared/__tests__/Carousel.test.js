import {Carousel, CarouselItem} from "../Carousel"

global.scrollTo = jest.fn()

describe("Carousel component", () => {
  const props = {galleries: {}, match: {path: "/path"}}
  const wrapper = shallow(<Carousel {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1)
  })

  it("scrolled to top", () => {
    expect(scrollTo).toBeCalledWith(0,0)
  })

  it("section gets right id", () => {
    expect(wrapper.find("section").prop("id")).toBe(props.match.path.replace("/", ""))
  })
})


describe("CarouselItem component", () => {
  const props = {
    className: "className",
    title: "title",
    desc: "desc",
    SIZE_640: "SIZE_640.jpg",
    SIZE_1024: "SIZE_1024.jpg",
    SIZE_1440: "SIZE_1440.jpg",
    SIZE_ORIGINAL: "SIZE_ORIGINAL.jpg",
    itemClassName: "fake-classname"
  }
  const wrapper = shallow(<CarouselItem {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("inherits classname", () => {
    expect(wrapper.prop("classname")).toBe(props.classname)
  })

  it("title shown", () => {
    expect(wrapper.find("h4").prop("children")).toBe(props.title)
  })

  it("description shown", () => {
    expect(wrapper.find("p").prop("children")).toBe(props.desc)
  })

  describe("source images are defined", () => {
    it("<640px", () => {
      expect(wrapper.findWhere(e => e.prop("srcSet") === props.SIZE_640).length).toBe(1)
    })
    it("<1280px", () => {
      expect(wrapper.findWhere(e => e.prop("srcSet") === props.SIZE_1024).length).toBe(1)
    })
    it("<1920px", () => {
      expect(wrapper.findWhere(e => e.prop("srcSet") === props.SIZE_1440).length).toBe(1)
    })

    it("default image", () => {
      expect(wrapper.find(".carousel-item-img img").prop("src")).toBe(props.SIZE_ORIGINAL)
    })
  })

})