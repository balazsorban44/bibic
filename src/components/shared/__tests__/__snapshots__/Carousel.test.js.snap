// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CarouselItem
    SIZE_1024="SIZE_1024.jpg"
    SIZE_1440="SIZE_1440.jpg"
    SIZE_640="SIZE_640.jpg"
    SIZE_ORIGINAL="SIZE_ORIGINAL.jpg"
    className="className"
    desc="desc"
    itemClassName="fake-classname"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          <img
            alt="blurred background"
            className="carousel-item-bg"
            src="SIZE_640.jpg"
          />
        </span>,
        <div
          className="carousel-item-picture"
        >
          <picture
            className="carousel-item-img"
          >
            <source
              media="(max-width: 640px)"
              srcSet="SIZE_640.jpg"
            />
            <source
              media="(max-width: 1024px)"
              srcSet="SIZE_1024.jpg"
            />
            <source
              media="(max-width: 1440px)"
              srcSet="SIZE_1440.jpg"
            />
            <img
              alt="title"
              src="SIZE_ORIGINAL.jpg"
            />
          </picture>
        </div>,
        <div
          className="carousel-item-desc"
        >
          <h4>
            title
          </h4>
          <p>
            desc
          </p>
        </div>,
      ],
      "className": "className",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="blurred background"
            className="carousel-item-bg"
            src="SIZE_640.jpg"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "blurred background",
            "className": "carousel-item-bg",
            "src": "SIZE_640.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <picture
            className="carousel-item-img"
          >
            <source
              media="(max-width: 640px)"
              srcSet="SIZE_640.jpg"
            />
            <source
              media="(max-width: 1024px)"
              srcSet="SIZE_1024.jpg"
            />
            <source
              media="(max-width: 1440px)"
              srcSet="SIZE_1440.jpg"
            />
            <img
              alt="title"
              src="SIZE_ORIGINAL.jpg"
            />
          </picture>,
          "className": "carousel-item-picture",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <source
                media="(max-width: 640px)"
                srcSet="SIZE_640.jpg"
              />,
              <source
                media="(max-width: 1024px)"
                srcSet="SIZE_1024.jpg"
              />,
              <source
                media="(max-width: 1440px)"
                srcSet="SIZE_1440.jpg"
              />,
              <img
                alt="title"
                src="SIZE_ORIGINAL.jpg"
              />,
            ],
            "className": "carousel-item-img",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(max-width: 640px)",
                "srcSet": "SIZE_640.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(max-width: 1024px)",
                "srcSet": "SIZE_1024.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(max-width: 1440px)",
                "srcSet": "SIZE_1440.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "title",
                "src": "SIZE_ORIGINAL.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "picture",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              title
            </h4>,
            <p>
              desc
            </p>,
          ],
          "className": "carousel-item-desc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "desc",
            },
            "ref": null,
            "rendered": "desc",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <img
              alt="blurred background"
              className="carousel-item-bg"
              src="SIZE_640.jpg"
            />
          </span>,
          <div
            className="carousel-item-picture"
          >
            <picture
              className="carousel-item-img"
            >
              <source
                media="(max-width: 640px)"
                srcSet="SIZE_640.jpg"
              />
              <source
                media="(max-width: 1024px)"
                srcSet="SIZE_1024.jpg"
              />
              <source
                media="(max-width: 1440px)"
                srcSet="SIZE_1440.jpg"
              />
              <img
                alt="title"
                src="SIZE_ORIGINAL.jpg"
              />
            </picture>
          </div>,
          <div
            className="carousel-item-desc"
          >
            <h4>
              title
            </h4>
            <p>
              desc
            </p>
          </div>,
        ],
        "className": "className",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="blurred background"
              className="carousel-item-bg"
              src="SIZE_640.jpg"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "blurred background",
              "className": "carousel-item-bg",
              "src": "SIZE_640.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <picture
              className="carousel-item-img"
            >
              <source
                media="(max-width: 640px)"
                srcSet="SIZE_640.jpg"
              />
              <source
                media="(max-width: 1024px)"
                srcSet="SIZE_1024.jpg"
              />
              <source
                media="(max-width: 1440px)"
                srcSet="SIZE_1440.jpg"
              />
              <img
                alt="title"
                src="SIZE_ORIGINAL.jpg"
              />
            </picture>,
            "className": "carousel-item-picture",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <source
                  media="(max-width: 640px)"
                  srcSet="SIZE_640.jpg"
                />,
                <source
                  media="(max-width: 1024px)"
                  srcSet="SIZE_1024.jpg"
                />,
                <source
                  media="(max-width: 1440px)"
                  srcSet="SIZE_1440.jpg"
                />,
                <img
                  alt="title"
                  src="SIZE_ORIGINAL.jpg"
                />,
              ],
              "className": "carousel-item-img",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(max-width: 640px)",
                  "srcSet": "SIZE_640.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(max-width: 1024px)",
                  "srcSet": "SIZE_1024.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(max-width: 1440px)",
                  "srcSet": "SIZE_1440.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "title",
                  "src": "SIZE_ORIGINAL.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "picture",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                title
              </h4>,
              <p>
                desc
              </p>,
            ],
            "className": "carousel-item-desc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
              },
              "ref": null,
              "rendered": "title",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "desc",
              },
              "ref": null,
              "rendered": "desc",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
